mail/spring-cloud-starter-stream-source-mail/src/main/java/org/springframework/cloud/stream/app/mail/source/MailSourceConfiguration.java

```diff
@@ -0,0 +1,201 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.mail.source;
+
+import java.util.Properties;
+
+import javax.mail.URLName;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.Bindings;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.app.trigger.TriggerConfiguration;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlowBuilder;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.integration.dsl.SourcePollingChannelAdapterSpec;
+import org.springframework.integration.dsl.mail.Mail;
+import org.springframework.integration.dsl.mail.MailInboundChannelAdapterSpec;
+import org.springframework.integration.dsl.support.Consumer;
+import org.springframework.integration.dsl.support.Transformers;
+import org.springframework.integration.scheduling.PollerMetadata;
+
+/**
+ * A source module that listens for mail and emits the content as a message payload.
+ *
+ * @author Amol
+ */
+@EnableBinding(Source.class)
+@EnableConfigurationProperties({ MailSourceProperties.class })
+@Import({ TriggerConfiguration.class })
+public class MailSourceConfiguration {
+
+	@Autowired
+	@Qualifier("defaultPoller")
+	PollerMetadata defaultPoller;
+
+	@Autowired
+	@Bindings(MailSourceConfiguration.class)
+	Source source;
+
+	@Autowired
+	MailSourceProperties properties;
+
+	@Bean
+	public IntegrationFlow mailInboundFlow() {
+
+		IntegrationFlowBuilder flowBuilder;
+
+		flowBuilder = getFlowBuilder();
+
+		return flowBuilder.transform(Transformers.fromMail()).channel(source.output())
+				.get();
+	}
+
+	/**
+	 * Method to build Integration Flow for Mail. Suppress Warnings for
+	 * MailInboundChannelAdapterSpec.
+	 * @return Integration Flow object for Mail Source
+	 */
+	@SuppressWarnings({ "rawtypes", "unchecked" })
+	private IntegrationFlowBuilder getFlowBuilder() {
+
+		IntegrationFlowBuilder flowBuilder;
+		URLName urlName;
+		if (null != this.properties.getMailUrl()) {
+			urlName = new URLName(this.properties.getMailUrl());
+		}
+		else {
+			urlName = null;
+		}
```

Comment:
This `if...else` logic around `null` is redundant. You already said that `mailUrl` that is it `@NotNull`.
On the other hand the `@ConfigurationProperties` is enough magic with the `ConversionService` behind and we can just have it like:
```
private URLName mailUrl;
...
@NotNull
public URLName getMailUrl() {
	return mailUrl;
}

...
public void setMailUrl(URLName mailUrl) {
	this.mailUrl = mailUrl;
}
```
in the `MailSourceProperties`. And don't worry about this string conversions manually.

In other words: that simplifies our life! :smile: